File: Transaction.java
public class Transaction {
    public String type;
    public double amount;
    public String category;
    public String date;

    public Transaction() {} // Required for Firebase

    public Transaction(String type, double amount, String category, String date) {
        this.type = type;
        this.amount = amount;
        this.category = category;
        this.date = date;
    }
}

File: Transaction.kt
package com.example.financetracker.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "transactions")
data class Transaction(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val title: String,               // e.g. "Salary" or "Groceries"
    val amount: Double,
    val type: String,                // "income" or "expense"
    val category: String,            // e.g. "Food", "Transport"
    val timestamp: Long = System.currentTimeMillis(),
    val note: String? = null         // Optional user note
)
